### DISCUSSION TITLE
Is it possible to retrieve the confirmation_token in auth schema for custom Resend emails?

### DISCUSSION THREAD

**Discussion by Spherrrical** [supabase/supabase#20401] [Questions] [answered]
Hi everyone,

I'm currently trying to use Resend as my email provider, instead of Supabase, for React styling, but I'm stuck on creating a "verify email" email. I've looked thru the email templates, and it seems like the only way to retrieve the confirmation_token is with Supabase's email templates, however it's very plain and I want to move away from it. Then, I tried seeing if I could fetch the confirmation token with the getUser object, but there is no such parameter that allows me (ex. data.user?.confirmation_token doesnt exist). It would highly appreciated if I could find a workaround to this, if not, I'll stick to using Supabase's email templates :)

Thanks,
Musa

**Comment by GaryAustin1** [comment, marked as answer]
You can use Resend as the SMTP provider and then the templates go thru them.  You can format the templates a bit with html/css.

If you want to totally send your own emails you can use https://supabase.com/docs/reference/javascript/auth-admin-generatelink to get urls and tokens to put in your own email.  You do this serverside with service_role key.

**Reply by Spherrrical** [reply]
Didn't catch that reference in the admin section, probably didnt look hard enough :) Thanks for the response!
